File: swift/bob/Sources/Bob/Bob.swift
//Solution goes in Sources

class Bob {

    public class func hey(_ input: String) -> String {

        let nospace = input.filter {
            !$0.isWhitespace
        }

        if nospace.isEmpty {
            return "Fine. Be that way!"
        }

        if !nospace.filter{ $0.isUppercase }.isEmpty && nospace.filter{ $0.isLowercase }.isEmpty {

//            if nospace.last! == "?" {
//                return "Calm down, I know what I'm doing!"
//            }

            return "Whoa, chill out!"

        }
        if nospace.last! == "?" {
            return "Sure."
        }


        return "Whatever."

    }
}

File: vbnet/bob/Bob.vb
Public Class Bob
    Public Function Hey(statement As String) As String
        Dim isQuestion = statement.EndsWith("?")
        Dim isUppercase = Text.RegularExpressions.Regex.Match(statement, "[A-Z]").Success AndAlso
                          (Not Text.RegularExpressions.Regex.Match(statement, "[a-z]").Success)
        Dim isEmpty = statement.Trim() = String.Empty
        If isEmpty Then
            Return "Fine. Be that way!"
        ElseIf isUppercase Then
            Return If (isQuestion, "Calm down, I know what I'm doing!", "Whoa, chill out!")
        End If
        Return If (isQuestion, "Sure.", "Whatever.")
    End Function
End Class
